---
title:    "Growth factor receptor network (GFRN) characterization in breast cancer"
purpose:  Performs all analyses and generates figures for the Manuscript "Discrete breast cancer growth and survival phenotypes influence apoptosis and drug response"
name:     GRFN_characterization_in_brest_cancer_Final.Rmd
authors:  Mumtahena Rahman, David Jenkins, Shelley MacNeil
date:     18April2016,update:20April2016, update:23May2016, update:6June16, update:15June16, update:30June16, update:06Nov16, update:18Dec16

---
# 2(A)-(D): Analysis of pathway activity and intrinsic subtypes
##2(A): TCGA BRCA all pathway
```{r cache=TRUE,echo=FALSE}

#Load required packages
if (!require("gplots")) {
   install.packages("gplots", dependencies = TRUE)
   library(gplots)
   }

if (!require("RColorBrewer")) {
   install.packages("RColorBrewer", dependencies = TRUE)
   library(RColorBrewer)
}
if (!require("data.table")) {
   install.packages("data.table", dependencies = TRUE)
   library(data.table)
}
if (!require("mclust")) {
   install.packages("mclust", dependencies = TRUE)
   library(mclust)
}
if (!require("ggplot2")) {
   install.packages("ggplot2", dependencies = TRUE)
   library(ggplot2)
}
if (!require("gridExtra")) {
   install.packages("gridExtra", dependencies = TRUE)
   library(gridExtra)
}

if (!require("devtools")) {
   install.packages("devtools", dependencies = TRUE)
   library(devtools)
  #install development version of ComplexHeatmap https://github.com/jokergoo/ComplexHeatmap
   install_github("jokergoo/ComplexHeatmap")  
   library(ComplexHeatmap)
}
   library(ComplexHeatmap)
if (!require("RColorBrewer")) {
   install.packages("RColorBrewer", dependencies = TRUE)
   library(RColorBrewer)
}
if (!require("cluster")) {
   install.packages("cluster", dependencies = TRUE)
   library(cluster)
}  
#souce key functions
source('Key_ASSIGN_functions_balancedsig.R', echo=FALSE)

# SHELLLEY'S REMOVE
source('~/Documents/PhDProjects/Multipathway_Modeling/GFRN_signatures/Key_ASSIGN_functions_balancedsig.R', echo=FALSE)

# set colors for heatmap
my_palette <- colorRampPalette(c("darkblue","aliceblue","brown4"))(n = 299)
col_breaks = c(seq(0,0.2,length=100), seq(0.2,0.4,length=100), seq(0.4,1,length=100)) 

#Specify input files (modify these locations for your system)

signatures_dir <- "Datasets"

#shelley's delete
signatures_dir <- "~/Documents/PhDProjects/Multipathway_Modeling/Data/"


# ICBP files
icbp_gene_expression<-paste(signatures_dir,"icbp_Rsubread_tpmlog.txt",sep="/")
icbp_drug_response<-paste(signatures_dir,"ICBP_drugs.txt",sep="/")
single_pathway_best_icbp_file<-paste(signatures_dir,"optimized_single_pathway_icbp_scaled.txt",sep="/")

# TCGA files 

tcga_brca_tumor_gene_expression<-paste(signatures_dir,"PANCAN24_BRCA_1119_TPMlog2.txt",sep="/")
# downloaded from http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM1536837&format=file&file=GSM1536837%5F06%5F01%5F15%5FTCGA%5F24%2Etumor%5FRsubread%5FTPM%2Etxt%2Egz and filtered for only BRCA tumor samples
tcga_clinical<-paste(signatures_dir,"GSE62944_06_01_15_TCGA_24_548_Clinical_Variables_9264_Samples.txt",sep="/")
#downloaded from http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE62944&format=file&file=GSE62944%5F06%5F01%5F15%5FTCGA%5F24%5F548%5FClinical%5FVariables%5F9264%5FSamples%2Etxt%2Egz
tcga_brca_pam50<-paste(signatures_dir,"BRCA.547.PAM50.SigClust.Subtypes.txt",sep="/")
#downloaded from https://tcga-data.nci.nih.gov/docs/publications/brca_2012/BRCA.547.PAM50.SigClust.Subtypes.txt
single_pathway_best_tcga_file<-paste(signatures_dir,"optimized_single_pathway_tcga_scaled.txt",sep="/")
#Input files (Modify These to Your System's Locations)

icbp_status_file      <- file.path("Datasets", "icbpstatus_v2.txt")
icbp_intrinsic_file   <- file.path("Datasets", "intrinsic.txt")
tcga_status_rds_file  <- file.path("Datasets", "tcga_er_pr_her2_formatted.rds")
tcga_intrinsic_file   <- file.path("Datasets", "TCGA_single_pathway_best_phenotype_intrinsic.txt")##Do we need this file? We could

# Shelley Delete
icbp_status_file      <- paste(signatures_dir, "icbpstatus_v2.txt", sep="/")
icbp_intrinsic_file   <- paste(signatures_dir,  "intrinsic.txt", sep="/")
tcga_status_rds_file  <- paste(signatures_dir,  "tcga_er_pr_her2_formatted.rds", sep="/")
tcga_intrinsic_file   <-  paste(signatures_dir , "TCGA_single_pathway_best_phenotype_intrinsic.txt")##Do we need this file? We could


#input icbp pathway predictions
single_pathway_best_icbp <-read.table(single_pathway_best_icbp_file, stringsAsFactors=FALSE,header=1, row.names=1,sep='\t')
rownames(single_pathway_best_icbp)[50] <- "T47D.Kbluc"

#input tcga pathway predictions
single_pathway_best_tcga <-read.table(single_pathway_best_tcga_file, stringsAsFactors=FALSE,header=1, row.names=1,sep='\t')

##Read in the ICBP and TCGA datas

# ICBP Drug Response Data
drugs<-read.delim(icbp_drug_response, header=1, sep='\t',row.names=1)

# ICBP RNA-seq Data
icbp<-as.matrix(read.table(icbp_gene_expression, sep='\t', stringsAsFactors=FALSE, header=1, row.names=1))

#IHC Subtypes ICBP
icbp_status <- read.table(icbp_status_file, row.names = 1, header=T, sep="\t")

#Intrinsic Subtypes ICBP
intrinsic <- read.table(icbp_intrinsic_file, row.names = 1, header=T, sep="\t")
colnames(intrinsic) <- "Intrinsic.Subtype"
rownames(intrinsic)[50] <- "T47D.Kbluc"
intrinsic <- intrinsic[rownames(single_pathway_best_icbp),,drop=F]
intrinsic <- cbind(icbp_status, intrinsic)

# independent drug response assay (bild lab)
assay_ec50<-read.table(paste(signatures_dir,"Drug_response_assay.txt",sep="/"),header=1,row.names=1,sep='\t')

# Breast Cancer TCGA gene expression data
test<-data.frame(fread(tcga_brca_tumor_gene_expression), check.names=F,row.names=1)

#TCGA RPPA Data
prot_brca<-read.csv(paste(signatures_dir,"TCGA-BRCA-RBN.csv",sep="/"),header=1,check.names = F)

# TCGA BRCA PAM50 (intrinsic subtypes)
pam50<-read.table(tcga_brca_pam50,sep='\t', stringsAsFactors = T,header=T, row.names=1,check.names = F)

# TCGA BRCA clincal subtypes
clinicals<-data.frame(fread(tcga_clinical), check.names=F,row.names=1)

#color pallete
my_palette <- colorRampPalette(c("darkblue","aliceblue","brown4"))(n = 299)

#top annotation
phenotype.annotation <- data.frame(Phenotype=c("Survival Phenotype",
                                               "Growth Phenotype",
                                               "Growth Phenotype",
                                               "Survival Phenotype",
                                               "Survival Phenotype",
                                               "Growth Phenotype",
                                               "Growth Phenotype",
                                               "Growth Phenotype"))
topha <- HeatmapAnnotation(df = phenotype.annotation,
                           col = list(Phenotype = c("Survival Phenotype" =  "coral3", "Growth Phenotype" = "aquamarine4")),height = unit(0.333, "cm"))


#Intrinsic Subtypes TCGA
tcgaintrinsic <- read.table(tcga_intrinsic_file, 
                            row.names = 1, header=T, sep="\t")
tcgaintrinsic <- tcgaintrinsic[rownames(single_pathway_best_tcga),12,drop=F]
tcgaintrinsic$PAM50 <- as.character(tcgaintrinsic$PAM50)
rownames(tcgaintrinsic) <- rownames(single_pathway_best_tcga)
tcgaintrinsic$PAM50[is.na(tcgaintrinsic$PAM50)] <- "Unavailable"
tcgaintrinsic <- cbind(er_pr_her2_status2, tcgaintrinsic)

#IHC Subtypes TCGA
er_pr_her2_status2 <- readRDS(tcga_status_rds_file)


# #Call the phenotypes in TCGA (DO WE NEED THIS?)
# 
# 
# # scale TCGA pathway predictions
# single_pathway_best_tcga_scaled<- scale(single_pathway_best_tcga[,1:7],scale = T,center = T)
# single_pathway_best_tcga_scaled=as.data.frame(single_pathway_best_tcga_scaled)
# single_pathway_best_tcga_scaled$phenotype<-NA
# 
# # take the mean of the signature groups
# growth_tcga<-apply(single_pathway_best_tcga_scaled[,c(1,4,5)],1,mean)
# survival_tcga<-apply(single_pathway_best_tcga_scaled[,c(2,3,6,7)],1,mean)
# 
# for(i in 1:nrow(single_pathway_best_tcga_scaled))
# { 
#   if(growth_tcga[[i]] > survival_tcga[[i]]){
#     single_pathway_best_tcga_scaled[i,8] <-"Growth Phenotype"
#   }
#   else{
#     single_pathway_best_tcga_scaled[i,8] <- "Survival Phenotype"
# 
#   }
# }
# 

```

## 2(A): TCGA 
```{r}
ha_row_tcga2 = HeatmapAnnotation(df = tcgaintrinsic,
                                col = list(PR.Status = c("Positive" =  "#4DAF4A",
                                                         "Negative" = "#984EA3",
                                                         "Indeterminate" = "black",
                                                         "Unavailable" = "grey"),
                                           HER2.Status = c("Positive" =  "#FFFF33",
                                                           "Negative" = "#F781BF",
                                                           "Indeterminate" = "black",
                                                           "Equivocal" = "skyblue",
                                                           "Unavailable" = "grey"),
                                           ER.Status = c("Positive" =  "#E41A1C",
                                                         "Negative" = "#377EB8",
                                                         "Indeterminate" = "black",
                                                         "Unavailable" = "grey"),
                                           PAM50 = c("LumA" = brewer.pal(6, "Dark2")[5],
                                                     "LumB" = brewer.pal(6, "Dark2")[2],
                                                     "Her2" = brewer.pal(6, "Dark2")[4],
                                                     "Basal" = brewer.pal(6, "Dark2")[1],
                                                     "Normal" = brewer.pal(6, "Dark2")[6],
                                                     "Unavailable"="grey")),
                                which = "row", width = unit(1.333, "cm"), show_legend = T)

h1 <- Heatmap(scale(single_pathway_best_tcga[,c(1:7)]), cluster_rows = T, cluster_columns = T,
              show_row_names = F, show_column_names = T, row_title_gp = gpar(fontsize =10),
              combined_name_fun = NULL,col=my_palette, top_annotation = topha,
              name="Scaled\nPathway\nActivity", column_title = "TCGA BRCA", show_heatmap_legend = T,
              column_dend_reorder = c(1,100,100,10,1,100,100))

#pdf("tcga.pdf", width=6)
draw(h1+ha_row_tcga2,row_dend_side = "left", annotation_legend_side = "bottom")
#dev.off()


```

## 2(B): ICBP all pathways

```{r echo=FALSE, warning=FALSE, message=FALSE, fig.width=7, fig.height=7,cache=TRUE}
set.seed(123)

ha_row_icbp1 = HeatmapAnnotation(df = intrinsic,
                                 name="Intrinsic Subtype",
                                 col = list(PR = c("Positive" =  "#4DAF4A",
                                                   "Negative" = "#984EA3",
                                                   "Unavailable" = "grey"),
                                       HER2 = c("Positive" =  "#FFFF33",
                                                "Negative" = "#F781BF",
                                                "Unavailable" = "grey"),
                                       ER = c("Positive" =  "#E41A1C",
                                              "Negative" = "#377EB8",
                                              "Unavailable" = "grey"),
                                       Intrinsic.Subtype = c("Basal" = brewer.pal(6, "Dark2")[1],
                                                             "Claudin-low" = brewer.pal(6, "Dark2")[2],
                                                             "HER2-Basal" = brewer.pal(6, "Dark2")[3],
                                                             "HER2-Luminal" = brewer.pal(6, "Dark2")[4],
                                                             "Luminal" = brewer.pal(6, "Dark2")[5],
                                                             "Normal-like" = brewer.pal(6, "Dark2")[6],
                                                             "Unavailable" = "grey")),
                                 which = "row", width = unit(1.333, "cm"), show_legend = T)

h1 <- Heatmap(scale(single_pathway_best_icbp[,c(1:7)]), cluster_rows = T, cluster_columns = T,
              show_row_names = F, show_column_names = T, row_title_gp = gpar(fontsize =10),
              combined_name_fun = NULL, name="Scaled\nPathway\nActivity",col=my_palette,
              top_annotation = topha, column_title = "ICBP", show_heatmap_legend = T,
              column_dend_reorder = c(1,100,100,10,1,100,100))


draw(h1+ha_row_icbp1,row_dend_side = "left", annotation_legend_side = "bottom")

```

## 2(C): TCGA, k-means of 4 pathways

```{r echo=FALSE, warning=FALSE, message=FALSE, fig.width=7, fig.height=7,cache=TRUE}
set.seed(123)
kmeans.4 <- kmeans(scale(single_pathway_best_tcga)[,c(1,2,3,4)], 4, iter.max=10000, nstart = 1000)
kmeans.4.clusters.tcga <-data.frame(kmeans.4$cluster)
kmeans.4.clusters.tcga[kmeans.4.clusters.tcga$kmeans.4.cluster == "3",] <- "Survival/HER2 high"
kmeans.4.clusters.tcga[kmeans.4.clusters.tcga$kmeans.4.cluster == "1",] <- "Survival/HER2 low"
kmeans.4.clusters.tcga[kmeans.4.clusters.tcga$kmeans.4.cluster == "2",] <- "Growth/BAD low"
kmeans.4.clusters.tcga[kmeans.4.clusters.tcga$kmeans.4.cluster == "4",] <- "Growth/BAD high"

colnames(kmeans.4.clusters.tcga) <- "KMeans.Clusters"
ha_row_tcga = HeatmapAnnotation(df =single_pathway_best_tcga,
                                col = list(KMeans.Clusters = c("Survival/HER2 high" = brewer.pal(5, "Set1")[1],
                                                                "Survival/HER2 low" = brewer.pal(5, "Set1")[2],
                                                                "Growth/BAD high" = brewer.pal(5, "Set1")[3],
                                                                "Growth/BAD low" = brewer.pal(5, "Set1")[5])),
                                which = "row", width = unit(0.333, "cm"), show_legend = T)
ha_row_tcga2 = HeatmapAnnotation(df = tcgaintrinsic,
                                col = list(PR.Status = c("Positive" =  "#4DAF4A",
                                                         "Negative" = "#984EA3",
                                                         "Indeterminate" = "black",
                                                         "Unavailable" = "grey"),
                                           HER2.Status = c("Positive" =  "#FFFF33",
                                                           "Negative" = "#F781BF",
                                                           "Indeterminate" = "black",
                                                           "Equivocal" = "skyblue",
                                                           "Unavailable" = "grey"),
                                           ER.Status = c("Positive" =  "#E41A1C",
                                                         "Negative" = "#377EB8",
                                                         "Indeterminate" = "black",
                                                         "Unavailable" = "grey"),
                                           PAM50 = c("LumA" = brewer.pal(6, "Dark2")[5],
                                                     "LumB" = brewer.pal(6, "Dark2")[2],
                                                     "Her2" = brewer.pal(6, "Dark2")[4],
                                                     "Basal" = brewer.pal(6, "Dark2")[1],
                                                     "Normal" = brewer.pal(6, "Dark2")[6],
                                                     "Unavailable"="grey")),
                                which = "row", width = unit(1.333, "cm"), show_legend = T)

h1 <- Heatmap(scale(single_pathway_best_tcga[,c(1:4)]), cluster_rows = T, cluster_columns = T,
              show_row_names = F, show_column_names = T,
              split=factor(kmeans.4.clusters.tcga$KMeans.Clusters,
                           levels=c("Survival/HER2 high","Survival/HER2 low","Growth/BAD high","Growth/BAD low")),
              row_title_gp = gpar(fontsize =10), column_dend_reorder = c(1,4,100,2),
              combined_name_fun = NULL,col=my_palette, top_annotation = topha,
              name="Scaled\nPathway\nActivity", column_title = "TCGA BRCA", show_heatmap_legend = T)

draw(ha_row_tcga+h1+ha_row_tcga2,row_dend_side = "left", annotation_legend_side = "bottom")

```

## 2(D): ICBP, k-means of 4 pathways

```{r echo=FALSE, warning=FALSE, message=FALSE, fig.width=7, fig.height=7,cache=TRUE}
set.seed(123)
kmeans.4 <- kmeans(scale(single_pathway_best_icbp)[,c(1,2,3,4)], 4, iter.max=100000, nstart = 100)
kmeans.4.clusters <-data.frame(kmeans.4$cluster)
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "3",] <- "Growth/BAD low"
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "1",] <- "Growth/BAD high"
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "4",] <- "Survival/HER2 high"
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "2",] <- "Survival/HER2 low"
colnames(kmeans.4.clusters) <- "KMeans.Clusters"
kmeans.4.clusters <- cbind(intrinsic, kmeans.4.clusters)

ha_row_icbp1 = HeatmapAnnotation(df = kmeans.4.clusters[,1:4,drop=F],
                                 name="Intrinsic Subtype",
                                 col = list(PR = c("Positive" =  "#4DAF4A",
                                                   "Negative" = "#984EA3",
                                                   "Unavailable" = "grey"),
                                       HER2 = c("Positive" =  "#FFFF33",
                                                "Negative" = "#F781BF",
                                                "Unavailable" = "grey"),
                                       ER = c("Positive" =  "#E41A1C",
                                              "Negative" = "#377EB8",
                                              "Unavailable" = "grey"),
                                       Intrinsic.Subtype = c("Basal" = brewer.pal(6, "Dark2")[1],
                                                             "Claudin-low" = brewer.pal(6, "Dark2")[2],
                                                             "HER2-Basal" = brewer.pal(6, "Dark2")[3],
                                                             "HER2-Luminal" = brewer.pal(6, "Dark2")[4],
                                                             "Luminal" = brewer.pal(6, "Dark2")[5],
                                                             "Normal-like" = brewer.pal(6, "Dark2")[6],
                                                             "Unavailable" = "grey")),
                                 which = "row", width = unit(1.333, "cm"), show_legend = T)

ha_row_icbp2 = HeatmapAnnotation(df = kmeans.4.clusters[,5,drop=F],
                                 name="K-Means Clusters",
                                col = list(KMeans.Clusters = c("Survival/HER2 high" = brewer.pal(5, "Set1")[1],
                                                                "Survival/HER2 low" = brewer.pal(5, "Set1")[2],
                                                                "Growth/BAD high" = brewer.pal(5, "Set1")[3],
                                                                "Growth/BAD low" = brewer.pal(5, "Set1")[5])),
                                which = "row", width = unit(0.333, "cm"), show_legend = T)

topha_4 <- HeatmapAnnotation(df = phenotype.annotation[c(1:4),, drop=F],
                        col = list(Phenotype = c("Survival Phenotype" =  "coral3",
                                                 "Growth Phenotype" = "aquamarine4")),
                        height = unit(0.333, "cm"))

h1 <- Heatmap(scale(single_pathway_best_icbp[,c(1,2,3,4)]), cluster_rows = T,
              cluster_columns = T, column_dend_reorder = c(1,4,4,1), show_row_names = F,
              show_column_names = T, split=factor(kmeans.4.clusters$KMeans.Clusters,
                           levels=c("Survival/HER2 high","Survival/HER2 low","Growth/BAD high","Growth/BAD low")),row_title_gp = gpar(fontsize =10),
              combined_name_fun = NULL, name="Scaled\nPathway\nActivity",col=my_palette,
              top_annotation = topha, column_title = "ICBP", show_heatmap_legend = T)


draw(ha_row_icbp2+h1+ha_row_icbp1, row_dend_side = "left", annotation_legend_side = "bottom")

```


# 3(A)-(C): Unsupervised variance analysis confirms the significance of the GFRN phenotypes
###Perform Principal Component Analysis in ICBP cell lines
```{r,cache=TRUE,echo=FALSE}
library(ggplot2)

# Remove genes with low variation (filtering)
dim(icbp)
icbp_f <-icbp[apply(icbp[,1:55]==0,1,mean) < 0.85,]
dim(icbp_f)

# Perfrom PCA Analysis 
pca_mat_icbp <- prcomp(t(icbp_f), center=T,scale=T)
rownames(pca_mat_icbp$x)[1:7]<-gsub("X","",rownames(pca_mat_icbp$x)[1:7])

# Plot ICBP PCA Results
#plot(pca_mat_icbp,main="ICBP")
cum_icbp<-cumsum((pca_mat_icbp$sdev)^2) / sum(pca_mat_icbp$sdev^2) #calculates cumulative proportion of variances
print(paste("Proportion of variance contributed by first 5 principal components in ICBP gene expression data:",cum_icbp[5]),sep=" ")#Proportion of variance contributed by first 5 principal component
#correlate the PCAs with GFRN pathway predictions from ASSIGN
icbp_pca_sig_cors=cor(pca_mat_icbp$x,single_pathway_best_icbp[,1:7],method="spearman")

#heatmap.2(as.matrix(icbp_pca_sig_cors[1:5,1:7]),col=my_palette,margins = c(9,7), trace= "none", main="ICBP PC/Signature Cors")

## Commented out the plots we didnt show. 


```

##3(A): Principal Component Analysis in TCGA BRCA
```{r,echo=FALSE}

# Filter for low varying genes
#dim(test) #23368
test_f<-test[apply(test[,1:ncol(test)]==0,1,mean) < 0.85,]
#dim(test_f) #21307

# Run PCA analysis in TCGA
pca_mat <- prcomp(t(test_f), center=T,scale=T)
cum_tcga<-cumsum((pca_mat$sdev)^2) / sum(pca_mat$sdev^2) #calculates cumulative proportiono of variances
print(paste("Proportion of variance contributed by first 5 principal components in TCGA BRCA gene expression data:",cum_tcga[5]),sep=" ")#Proportion of variance contributed by first 5 principal components

# Plot TCGA PCA results
plot(pca_mat,main="PCA in TCGA BRCA")


```

##3(B): Boxplot TCGA Principal Components across clinical subtypes (IHC: ER, PR, HER2)
```{r,echo=FALSE}

# prepare data
er_pr_her2_status<-t(clinicals[c("er_status_by_ihc","pr_status_by_ihc","her2_status_by_ihc"),])
er_pr_her2_status<- er_pr_her2_status[3:nrow(er_pr_her2_status),]

# merge PCAs with IHC
com_tcga_er_pr_her2<-merge_drop(pca_mat$x,er_pr_her2_status)

# assign HER, ER, PR positive and negative
er_pos<-com_tcga_er_pr_her2[com_tcga_er_pr_her2$er_status_by_ihc=="Positive",]
er_neg<-com_tcga_er_pr_her2[com_tcga_er_pr_her2$er_status_by_ihc=="Negative",]
pr_pos<-com_tcga_er_pr_her2[com_tcga_er_pr_her2$pr_status_by_ihc=="Positive",]
pr_neg<-com_tcga_er_pr_her2[com_tcga_er_pr_her2$pr_status_by_ihc=="Negative",]
her_pos<-com_tcga_er_pr_her2[com_tcga_er_pr_her2$her2_status_by_ihc=="Positive",]
her_neg<-com_tcga_er_pr_her2[com_tcga_er_pr_her2$her2_status_by_ihc=="Negative",]
##
pca_phenotypes<-merge_drop(pca_mat$x,single_pathway_best_tcga)
survival_phen<-pca_phenotypes[pca_phenotypes$phenotype=="Survival Phenotype",]
growth_phen<-pca_phenotypes[pca_phenotypes$phenotype=="Growth Phenotype",]

# Make boxplots for TCGA PCs across IHC subtypes
par(mfrow=c(3,2),lwd=2)

boxplot(er_pos$PC1,er_neg$PC1,pr_pos$PC1,pr_neg$PC1,her_pos$PC1,her_neg$PC1,survival_phen$PC1,growth_phen$PC1,las=2, at=c(1,2,4,5,7,8,10,11),names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main=paste("PC1:",round(cum_tcga[1]*100,digits = 2),"% of total variance",sep = " "), col=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"),ylab="PC1 values")
text(1.5,100,"*",col = "red",cex=4)
text(4.5,100,"*",col = "red",cex=4)
boxplot(er_pos$PC2,er_neg$PC2,pr_pos$PC2,pr_neg$PC2,her_pos$PC2,her_neg$PC2,survival_phen$PC2,growth_phen$PC2,las=2,at=c(1,2,4,5,7,8,10,11),col=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"),names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main=paste("PC2:",round((cum_tcga[2]-cum_tcga[1])*100,digits = 2),"% of total variance",sep = " "),ylab="PC2 values")
text(1.5,100,"*",col = "red",cex=4)
text(4.5,100,"*",col = "red",cex=4)
text(10.5,100,"*",col = "red",cex=4)

boxplot(er_pos$PC3,er_neg$PC3,pr_pos$PC3,pr_neg$PC3,her_pos$PC3,her_neg$PC3,survival_phen$PC3,growth_phen$PC3,top = T,las=2, at=c(1,2,4,5,7,8,10,11), names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main=paste("PC3:",round((cum_tcga[3]-cum_tcga[2])*100,digits = 2),"% of total variance",sep = " "), col=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"),ylab="PC3 values")
text(1.5,50,"*",col = "red",cex=4)
text(4.5,50,"*",col = "red",cex=4)
text(7.5,50,"*",col = "red",cex=4)
text(10.5,50,"*",col = "red",cex=4)

boxplot(er_pos$PC4,er_neg$PC4,pr_pos$PC4,pr_neg$PC4,her_pos$PC4,her_neg$PC4,survival_phen$PC4,growth_phen$PC4,top = T,las=2, at=c(1,2,4,5,7,8,10,11), names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main=paste("PC4:",round((cum_tcga[4]-cum_tcga[3])*100,digits = 2),"% of total variance",sep = " "), col=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"),ylab="PC4 values")
text(1.5,100,"*",col = "red",cex=4)
text(4.5,100,"*",col = "red",cex=4)
text(7.5,100,"*",col = "red",cex=4)
text(10.5,100,"*",col = "red",cex=4)

boxplot(er_pos$PC5,er_neg$PC5,pr_pos$PC5,pr_neg$PC5,her_pos$PC5,her_neg$PC5,survival_phen$PC5,growth_phen$PC5,top = T,las=2, at=c(1,2,4,5,7,8,10,11), names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main=paste("PC5:",round((cum_tcga[5]-cum_tcga[4])*100,digits = 2),"% of total variance",sep = " "), col=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"),ylab="PC5 values")
text(1.5,100,"*",col = "red",cex=4)
text(4.5,100,"*",col = "red",cex=4)
text(7.5,100,"*",col = "red",cex=4)
text(10.5,100,"*",col = "red",cex=4)

##Significance calculation
#PC1 comparisons:
t.test(er_pos$PC1,er_neg$PC1);t.test(pr_pos$PC1,pr_neg$PC1);t.test(her_pos$PC1,her_neg$PC1);t.test(survival_phen$PC1,growth_phen$PC1)
#PC2 comparisons:
t.test(er_pos$PC2,er_neg$PC2);t.test(pr_pos$PC2,pr_neg$PC2);t.test(her_pos$PC2,her_neg$PC2);t.test(survival_phen$PC2,growth_phen$PC2)
#PC3 comparisons:
t.test(er_pos$PC3,er_neg$PC3);t.test(pr_pos$PC3,pr_neg$PC3);t.test(her_pos$PC3,her_neg$PC3);t.test(survival_phen$PC3,growth_phen$PC3)
#PC4 comparisons:
t.test(er_pos$PC4,er_neg$PC4);t.test(pr_pos$PC4,pr_neg$PC4);t.test(her_pos$PC4,her_neg$PC4);t.test(survival_phen$PC4,growth_phen$PC4)
#PC5 comparisons:
t.test(er_pos$PC5,er_neg$PC5);t.test(pr_pos$PC5,pr_neg$PC5);t.test(her_pos$PC5,her_neg$PC5);t.test(survival_phen$PC5,growth_phen$PC5)



```

##3(C) Correlate TCGA Principal Components with ASSIGN generated GFRN pathway predictions
```{r,cache=TRUE,echo=FALSE}
library(gplots)

pca1_5=pca_mat$x[,1:5]
pca_pathway_cors= cor(pca1_5, single_pathway_best_tcga[,1:7], method = "spearman")

# Heatmap with signature/PC correlations 
heatmap.2(as.matrix(pca_pathway_cors),col=my_palette,main="Cor between PCs 1-5 and signatures in TCGA  \n BRCA samples", trace = "none", key=T)

heatmap.2(as.matrix(pca_pathway_cors),col=my_palette,trace="none",margins=c(7,7),main="",ColSideColors = c("coral3","aquamarine4","aquamarine4","coral3","coral3","aquamarine4","aquamarine4"), key.title="", key=T)
legend("bottomleft",legend = c("Survival", "Growth"), col = c("coral3","aquamarine4"),  lty= 1,lwd = 8,cex = .7,border = NULL)
```

#4(A)-(E):Survival and growth phenotypes express dichotomous cell survival mechanisms
##4(A):Heatmap of breast cancer cell lines from Western Blots
```{r,echo=FALSE,cache=TRUE}

#parse the cell lines
selected_celllines<-c("AU565","BT474","BT483","BT549","CAMA1","HCC1143","HCC1419","HCC1428","HCC1806","HCC1937","HCC1954","HCC38","HCC70","HS578T","JIMT1","MCF7","MDAMB175","MDAMB231","T47D","ZR751")
scaled_selected<-single_pathway_best_icbp[selected_celllines,]

#make the heatmap
par(mfrow=c(1,1),lwd=3)
heatmap.2(as.matrix(single_pathway_best_icbp[selected_celllines,c(1:7)]),col=my_palette,trace="none",margins=c(7,7),main="",ColSideColors = c("coral3","aquamarine4","aquamarine4","coral3","coral3","aquamarine4","aquamarine4"), ,key.title="",scale="row", key=T)#,Rowv=F,Colv=F)
legend("topright",legend = c("Survival phenotype", "Growth phenotype"), col = c("coral3","aquamarine4"),  lty= 1,lwd = 8,cex = 0.5)

# Heat map will the cell lines from old and new western blot. DELETE
#selected_celllines_both_assays<-c("AU565","BT474","BT483","BT549","CAMA1","HCC1143","HCC1419","HCC1428","HCC1806","HCC1937","HCC1954","HCC38","HCC70","HS578T","JIMT1","MCF7","MDAMB175","MDAMB231","T47D","ZR751", "HCC3153", "HCC1395", "21PT", "HCC2218", "HCC1569", "LY2", "MDAMB175", "MDAMB361", "SKBR3", "SUM53PE", "ZR75B", "ZR7530")
#length(selected_celllines_both_assays) #32
#scaled_selected_all<-single_pathway_best_icbp[selected_celllines_both_assays,]
#make the heatmap, all
#remove this
#par(mfrow=c(1,1),lwd=3)
#heatmap.2(as.matrix(single_pathway_best_icbp[,c(1:7)]),col=my_palette,trace="none",margins=c(7,7),main="",ColSideColors = c("coral3","aquamarine4","aquamarine4","coral3","coral3","aquamarine4","aquamarine4"), ,key.title="",scale="row", key=T)#,Rowv=F,Colv=F)
#legend("topright",legend = c("Survival phenotype", "Growth phenotype"), col = c("coral3","aquamarine4"),  lty= 1,lwd = 8,cex = 0.5)
```

##4(C)-(E): Gene and protein expression are significantly different between growth and survival phenotypes
```{r,echo=FALSE}

# pull the genes we are intereted in (BH3 apoptosis genes)    
exp<-as.data.frame(t(icbp[c("BCL2L11","MCL1"),]))
#exp<-as.data.frame(t(icbp[c("BAX","BCL2L11","BAD","BIK","PMAIP1","BBC3","BMF","BCL2","MCL1"),])) #DELETE
#rename rownames 
rownames(exp)[1:7]<-c("184A1","184B5","21MT1","21MT2","21NT","21PT","600MPE")

# merge ICBP expression data with phenotypes and subtypes
exp_best<-merge_drop(single_pathway_best_icbp[,1:8],exp)
exp_best_subtypes= merge_drop(exp_best, icbp_status)

# Boxplot MCL1 and BIM for the phenotypes (with the normals)
par(mfrow = c(1,1), lwd=4)
for (i in 9:ncol(exp_best)) {
  print(i)
  tmp = t.test(exp_best[,i] ~ exp_best[,8])
  boxplot(exp_best[,i] ~ exp_best[,8],
         main = paste(colnames(exp_best)[i],"\np-value* =",tmp$p.value), col= c("aquamarine4", "coral3"), ylab="log2 (Transcript per million)")
}

#Look for protein (RPPA) expression differences between apoptosis molecules in TCGA
# TCGA protein boxplots for phenotypes
# make TCGA names long
prot_brca$TCGA_patient_barcode<-short_to_long_TCGA_id(shortnames=gsub(pattern="\\.","-",as.character(prot_brca$TCGA_patient_barcode)),longnames=gsub(pattern="\\.","-",rownames(single_pathway_best_tcga)))

# merge protein data with pathway predictions
data_prot<-merge(single_pathway_best_tcga,prot_brca,by.x=0,by.y=1)

# pull out apoptoic proteins
#prot_of_interest<-c("BIM","BAD","MCL1")
prot_of_interest<-c("BIM","MCL1")
colnames(data_prot)<-toupper(colnames(data_prot))
colnames(data_prot)<-gsub(pattern = "-",replacement = "",colnames(data_prot))
data_prot_subtype<-merge(data_prot, er_pr_her2_status2, by.x=1,by.y=0)

# Boxblot the protein expression across phenotypes
#pdf("TCGA_apoptosis_boxplots_RPPA.pdf", width=6)
for(i in 1:length(prot_of_interest)){
  print(paste("analyzing",prot_of_interest[i]))
  if(prot_of_interest[i]%in%colnames(data_prot)){
    print(paste(prot_of_interest[i],"data available"))
    tmp<-t.test(data_prot[,prot_of_interest[i]]~data_prot$PHENOTYPE)
    par(lwd=4)
    boxplot(data_prot[,prot_of_interest[i]]~data_prot$PHENOTYPE,main=paste(prot_of_interest[i],"\np-value:",tmp$p.value,sep=' '), col= c("aquamarine4", "coral3"),yaxt= 'n', ylab="RPPA Score")   
    axis(2,cex.axis=2)
  }
  else{
    print("Sorry!This protein is unavailable in TCGA RPPA dataset!")
  }
}

#Look for gene expression differences between apoptosis molecules in TCGA
###TCGA gene expression based phenotype analysis
#exp_tcga<-t(test[c("BCL2L11","BAD","MCL1"),])
exp_tcga<-t(test_f[c("BCL2L11","MCL1"),])
exp_best<-merge_drop(single_pathway_best_tcga,exp_tcga)
tcga_exp_subtypes<-merge_drop(exp_best,er_pr_her2_status2 )

# Boxplots for TCGA gene exprssion between phenotypes
par(mfrow = c(1,1), lwd=4)
for (i in 10:ncol(exp_best)) {
  tmp = t.test(exp_best[,i] ~ exp_best[,8])
  boxplot(exp_best[,i] ~ exp_best[,8],main = paste(colnames(exp_best)[i],"\nTCGA p-value* =",tmp$p.value),col= c("aquamarine4", "coral3"))
}

#exp_best_subtypes_er_pos<-exp_best_subtypes[exp_best_subtypes$ER=="Positive",]
#exp_best_subtypes_er_neg<-exp_best_subtypes[exp_best_subtypes$ER=="Negative",]
#pdf("ICBP_apoptosis_boxplots_mRNA_ER.pdf", width=6)

#er_vs_er_neg_mcl=t.test(exp_best_subtypes_er_pos$MCL1,exp_best_subtypes_er_neg$MCL1)
#er_vs_er_neg_bim=t.test(exp_best_subtypes_er_pos$BCL2L11,exp_best_subtypes_er_neg$BCL2L11)

# Boxplot MCL1 and BIM for ER status
#boxplot(exp_best_subtypes_er_pos$MCL1, exp_best_subtypes_er_neg$MCL1, las=2, names=c("ER+","ER-"), col=c("coral3","aquamarine4", "#E41A1C","#377EB8"), ylab="log2 (Transcript per million)", main = paste("\nphenotype p-value* =",er_vs_er_neg_mcl$p.value, "\nER p-value = ", er_vs_er_neg_bim$p.value))
#boxplot(exp_best_subtypes_er_pos$BCL2L11, exp_best_subtypes_er_neg$BCL2L11, las=2, names=c("ER+","ER-"), col=c("coral3","aquamarine4", "#E41A1C","#377EB8"), ylab="log2 (Transcript per million)", main = paste( "\nER p-value = ", er_vs_er_neg_bim$p.value))
#dev.off()

## WITHOUT the Normals. 
# Same analysis without the normals
#pdf("~/Documents/PhDProjects/Multipathway_Modeling/Paper_Revisions/Apoptossis_Boxplots_ERstatus_noNormals_withBad.pdf", width=6)
#exp_best_subtypes_rm_nrml=exp_best_subtypes[-1,]
#exp_best_subtypes_rm_nrml=exp_best_subtypes_rm_nrml[-1,]
#exp_best_subtypes_rm_nrml=exp_best_subtypes_rm_nrml[-32,]
#exp_best_subtypes_rm_nrml=exp_best_subtypes_rm_nrml[-32,]
#exp_best_subtypes_rm_nrml=exp_best_subtypes_rm_nrml[-32,]

#mcl_icbp_t = t.test(exp_best_subtypes_rm_nrml$MCL1 ~ exp_best_subtypes_rm_nrml$phenotype)
#bim_icbp_t = t.test(exp_best_subtypes_rm_nrml$BCL2L11 ~ exp_best_subtypes_rm_nrml$phenotype)
#bad_icbp_t = t.test(exp_best_subtypes_rm_nrml$BAD.y ~ exp_best_subtypes_rm_nrml$phenotype)

#par(lwd=4)
#boxplot(exp_best_subtypes_rm_nrml$MCL1 ~ exp_best_subtypes_rm_nrml$phenotype, main = paste("\n ICBP MCL-1 p-value =", #mcl_icbp_t$p.value), col= c("aquamarine4", "coral3"), ylab="log2 (Transcript per million)")
#boxplot(exp_best_subtypes_rm_nrml$BCL2L11 ~ exp_best_subtypes_rm_nrml$phenotype, main = paste("\n ICBP BIM p-value* =", #bim_icbp_t$p.value), col= c("aquamarine4", "coral3"), ylab="log2 (Transcript per million)")
#boxplot(exp_best_subtypes_rm_nrml$BAD.y ~ exp_best_subtypes_rm_nrml$phenotype, main = paste("\n ICBP BAD p-value* =", #bad_icbp_t$p.value), col= c("aquamarine4", "coral3"), ylab="log2 (Transcript per million)")

# Boxplots, normals only, seperate based on ER Status
#exp_best_subtypes_er_pos_no_nrml<-exp_best_subtypes_rm_nrml[exp_best_subtypes_rm_nrml$ER=="Positive",]
#exp_best_subtypes_er_neg_no_nrml<-exp_best_subtypes_rm_nrml[exp_best_subtypes_rm_nrml$ER=="Negative",]
# t.test for MCL and BIM for ER status
#er_vs_er_neg_mcl_nn=t.test(exp_best_subtypes_er_pos_no_nrml$MCL1,exp_best_subtypes_er_neg_no_nrml$MCL1)
#er_vs_er_neg_bim_nn=t.test(exp_best_subtypes_er_pos_no_nrml$BCL2L11,exp_best_subtypes_er_neg_no_nrml$BCL2L11)
#er_vs_er_neg_bad_nn=t.test(exp_best_subtypes_er_pos_no_nrml$BAD.y,exp_best_subtypes_er_neg_no_nrml$BAD.y)
#par(lwd=4)
#boxplot(exp_best_subtypes_er_pos_no_nrml$MCL1, exp_best_subtypes_er_neg_no_nrml$MCL1, las=2, names=c("ER+","ER-"), col=c( "#E41A1C","#377EB8"), ylab="log2 (Transcript per million)", main = paste("\nICBP MCL-1 p-value* =",er_vs_er_neg_mcl_nn$p.value))
#boxplot(exp_best_subtypes_er_pos_no_nrml$BCL2L11, exp_best_subtypes_er_neg_no_nrml$BCL2L11, las=2, names=c("ER+","ER-"), col=c("#E41A1C","#377EB8"), ylab="log2 (Transcript per million)", main = paste( "\nICBP BIM p-value = ", er_vs_er_neg_bim_nn$p.value))
#boxplot(exp_best_subtypes_er_pos_no_nrml$BAD.y, exp_best_subtypes_er_neg_no_nrml$BAD.y, las=2, names=c("ER+","ER-"), col=c("#E41A1C","#377EB8"), ylab="log2 (Transcript per million)", main = paste( "\nICBP BAD p-value = ", er_vs_er_neg_bad_nn$p.value))

#exp_best_subtypes_growth<-exp_best_subtypes[exp_best_subtypes$phenotype=="Growth Phenotype",]
#exp_best_subtypes_survival<-exp_best_subtypes[exp_best_subtypes$phenotype=="Survival Phenotype",]

#MCL1
#par(mfrow = c(1,1), lwd=4)
#boxplot(exp_best_subtypes_survival$MCL1, exp_best_subtypes_growth$MCL1, exp_best_subtypes_er_pos$MCL1, exp_best_subtypes_er_neg$MCL1,las=2, names=c("Survival","Growth", "ER+","ER-"), col=c("coral3","aquamarine4", "#E41A1C","#377EB8"), ylab="log2 (Transcript per million)", main = paste("\nphenotype p-value* =",s_vs_g_mcl$p.value, "\nER p-value = ", er_vs_er_neg_mcl$p.value))
#BIM
#s_vs_g_bim = t.test(exp_best_subtypes_survival$BCL2L11,exp_best_subtypes_growth$BCL2L11)
#er_vs_er_neg_bim=t.test(exp_best_subtypes_growth_er_pos$BCL2L11,exp_best_subtypes_growth_er_pos$BCL2L11)
#par(mfrow = c(1,1), lwd=4)
#boxplot(exp_best_subtypes_survival$BCL2L11, exp_best_subtypes_growth$BCL2L11, exp_best_subtypes_er_pos$BCL2L11, exp_best_subtypes_er_neg$BCL2L11,las=2, names=c("Survival","Growth", "ER+","ER-"), col=c("coral3","aquamarine4", "#E41A1C","#377EB8"), ylab="log2 (Transcript per million)", main = paste("\nphenotype p-value* =",s_vs_g_bim$p.value, "\nER p-value = ", er_vs_er_neg_bim$p.value))

# ER status only ICBP gene expression
#par(mfrow = c(1,1), lwd=4)
#boxplot(exp_best_subtypes_er_pos$MCL1, exp_best_subtypes_er_neg$MCL1, exp_best_subtypes_er_pos$BCL2L11, exp_best_subtypes_er_neg$BCL2L11, las=2, names=c("ER+","ER-", "ER+","ER-"), col=c("coral3","aquamarine4", "#E41A1C","#377EB8"), ylab="log2 (Transcript per million)", main = paste("\nphenotype p-value* =",er_vs_er_neg_mcl$p.value, "\nER p-value = ", er_vs_er_neg_bim$p.value))

# RRPA seperate on ER status
#rppa_er_pos<-data_prot_subtype[data_prot_subtype$ER.Status=="Positive",]
#rppa_er_neg<-data_prot_subtype[data_prot_subtype$ER.Status=="Negative",]
#rppa_bim<-t.test(rppa_er_pos$BIM, rppa_er_neg$BIM)
#rppa_bad<-t.test(rppa_er_pos$BAD, rppa_er_neg$BAD)
#par(lwd=4)
#boxplot(rppa_er_pos$BIM,rppa_er_neg$BIM, main=paste("\nBIMp-value:",rppa_bim$p.value,sep=' '), col= c("#E41A1C","#377EB8"),yaxt= 'n', #ylab="RPPA Score", names=c("ER+","ER-"))  
#boxplot(rppa_er_pos$BIM,rppa_er_neg$BAD, main=paste("\nBAD p-value:",rppa_bad$p.value,sep=' '), col= c("#E41A1C","#377EB8"),yaxt= 'n', ylab="RPPA Score", names=c("ER+","ER-"))   

# TCGA gene expression: seperate on ER status
#tcga_exp_er_pos<-tcga_exp_subtypes[tcga_exp_subtypes$ER.Status=="Positive",]
#dim(tcga_exp_er_pos)
#tcga_exp_er_neg<-tcga_exp_subtypes[tcga_exp_subtypes$ER.Status=="Negative",]
#dim(tcga_exp_er_neg)
#tcga_exp_bim<-t.test(tcga_exp_er_pos$BCL2L11, tcga_exp_er_neg$BCL2L11)
#tcga_exp_mcl<-t.test(tcga_exp_er_pos$MCL1, tcga_exp_er_neg$MCL1)
#tcga_exp_bad<-t.test(tcga_exp_er_pos$BAD.y, tcga_exp_er_neg$BAD.y)
#par(lwd=4)
#boxplot(tcga_exp_er_pos$BCL2L1,tcga_exp_er_neg$BCL2L11, main=paste("TCGA BIM p-value:",tcga_exp_bim$p.value,sep=' '), col= c("#E41A1C","#377EB8"),yaxt= 'n', ylab="RPPA Score", names=c("ER+","ER-"))   
#boxplot(tcga_exp_er_pos$MCL1, tcga_exp_er_neg$MCL1, main=paste("TCGA MCL-1 p-value:",tcga_exp_bim$p.value,sep=' '), col= c("#E41A1C","#377EB8"),yaxt= 'n', ylab="RPPA Score", names=c("ER+","ER-"))   
#boxplot(tcga_exp_er_pos$BAD.y, tcga_exp_er_neg$BAD.y, main=paste("TCGA BAD p-value:",tcga_exp_bad$p.value,sep=' '), col= c("#E41A1C","#377EB8"),yaxt= 'n', ylab="RPPA Score", names=c("ER+","ER-"))  
#dev.off()

# western blot quantification
#New_Westerns_Quantifications <- read.delim("~/Documents/PhDProjects/Multipathway_Modeling/Paper_Revisions/New_Westerns_Quantifications.txt", row.names=1)
#Old_Westerns_Quantifications <- read.delim("~/Documents/PhDProjects/Multipathway_Modeling/Paper_Revisions/Old_Westerns_Quantifications.txt", row.names=1)

#mcl_new=t.test(New_Westerns_Quantifications$MCL1_new_norm ~ New_Westerns_Quantifications$phenotype)
#mcl_old=t.test(Old_Westerns_Quantifications$mcl_old_norm~ Old_Westerns_Quantifications$phenotype)
#bim_new=t.test(New_Westerns_Quantifications$BIM_new_norm ~ New_Westerns_Quantifications$phenotype)
#bim_old=t.test(Old_Westerns_Quantifications$bim_old_norm~ Old_Westerns_Quantifications$phenotype)
#bad_new=t.test(New_Westerns_Quantifications$BAD_new_norm ~ New_Westerns_Quantifications$phenotype)
#bad_old=t.test(Old_Westerns_Quantifications$bad_old_norm~ Old_Westerns_Quantifications$phenotype)
#
#pdf("~/Documents/PhDProjects/Multipathway_Modeling/Paper_Revisions/Westernblot_quantification_boxplots.pdf", width=10)
#par(lwd=4, mfrow = c(1, 2))
# MCL1 new
#boxplot(New_Westerns_Quantifications$MCL1_new_norm ~ New_Westerns_Quantifications$phenotype,  main = paste("\n MCL1 (new blot)", "\n", "p-value* =", round(mcl_new$p.value, digits=3)), col= c("aquamarine4", "coral3"), ylab="Density")
# MCL1 old
#boxplot(Old_Westerns_Quantifications$mcl_old_norm ~ Old_Westerns_Quantifications$phenotype,  main = paste("\n MCL-1 (old blot)", "\n", "p-value* =", round(mcl_old$p.value, digits=3)), col= c("aquamarine4", "coral3"), ylab="Density")
#Bim new
#par(lwd=4, mfrow = c(1, 2))
#boxplot(New_Westerns_Quantifications$BIM_new_norm ~ New_Westerns_Quantifications$phenotype,  main = paste("\n BIM (new blot)", "\n", "p-value* =", round(bim_new$p.value, digits=3)), col= c("aquamarine4", "coral3"), ylab="Density")
#Bim old
#boxplot(Old_Westerns_Quantifications$bim_old_norm ~ Old_Westerns_Quantifications$phenotype,  main = paste("\n BIM (old blot)", "\n", "p-value* =", round(bim_old$p.value, digits=3)), col= c("aquamarine4", "coral3"), ylab="Density)")
#bad new
#par(lwd=4, mfrow = c(1, 2))
#boxplot(New_Westerns_Quantifications$BAD_new_norm ~ New_Westerns_Quantifications$phenotype,  main = paste("\n BAD (new blot)", "\n", "p-value* =", round(bad_new$p.value, digits=3)), col= c("aquamarine4", "coral3"), ylab="Density")
#Bim old
#boxplot(Old_Westerns_Quantifications$bad_old_norm~ Old_Westerns_Quantifications$phenotype,  main = paste("\n BAD (old blot)", "\n", "p-value* =", round(bad_old$p.value, digits=3)), col= c("aquamarine4", "coral3"), ylab="Density")
#dev.off()
```

# 5(A) &(B): Growth factor network phenotypes reflect dichotomous drug response in breast cancer cell lines
##5(A): Correlate ICBP GFRN pathway predictions with ICBP drug response and make Heatmap
```{r, echo=FALSE,cache=TRUE}
library(gplots)
# Merge the ICBP GFRN pathway predictions with the ICBP drug response

colnames(drugs)[1:15]<-gsub("X","",colnames(drugs)[1:15])
pred_drug<-merge_drop(single_pathway_best_icbp,drugs,by.x = 0,by.y = 0,all.x =T)

cors_drugs=cor(pred_drug[,c("Sigma.AKT1.2.inhibitor","CGC.11047","Cisplatin","Docetaxel","Everolimus","GSK2119563","GSK2126458","GSK1059615", "GSK1120212", "GSK1059868","GSK2141795", "PF.4691502", "Paclitaxel","Rapamycin","Temsirolimus","Erlotinib","Gefitinib","AZD6244","AG1478","Lapatinib")],pred_drug[, 1:7],method="spearman",use="pairwise")
par(lwd=2)
heatmap.2(cors_drugs, col=my_palette, trace='none', density.info = 'none', margins = c(6,6), cexCol = 0.8, main="ICBP:spearman: row scaling", RowSideColors = c("pink","yellow","yellow","yellow","pink","pink","pink","pink","lightblue", "pink","pink", "pink", "yellow","pink","pink","lightblue","lightblue","lightblue","lightblue", "pink"), scale="row",ColSideColors = c("coral3","aquamarine4","aquamarine4","coral3","coral3","aquamarine4","aquamarine4"))
legend("bottomleft",legend = c("PI3K/AKT/mTOR inhibitors", "Chemotherapeutics","EGFR/MEK inhibitors","Growth phenotype","Survival phenotype"), col = c("pink","yellow","lightblue","coral3","aquamarine4"),  lty= 1,lwd = 10,cex = 1)

#seperate ER status

pred_drug_subtype=merge_drop(pred_drug,icbp_status)
colnames(pred_drug_subtype)
icbp_drug_er_pos=pred_drug_subtype[pred_drug_subtype$ER=="Positive",]

icbp_drug_er_neg=pred_drug_subtype[pred_drug_subtype$ER=="Negative",]

cors_drugs_er_pos=cor(icbp_drug_er_pos[,c("Sigma.AKT1.2.inhibitor","CGC.11047","Cisplatin","Docetaxel","Everolimus","GSK2119563","GSK2126458","GSK1059615", "GSK1120212", "GSK1059868","GSK2141795", "PF.4691502", "Paclitaxel","Rapamycin","Temsirolimus","Erlotinib","Gefitinib","AZD6244","AG1478","Lapatinib")],icbp_drug_er_pos[, 1:7],method="spearman",use="pairwise")

cors_drugs_er_neg=cor(icbp_drug_er_neg[,c("Sigma.AKT1.2.inhibitor","CGC.11047","Cisplatin","Docetaxel","Everolimus","GSK2119563","GSK2126458","GSK1059615", "GSK1120212", "GSK1059868","GSK2141795", "PF.4691502", "Paclitaxel","Rapamycin","Temsirolimus","Erlotinib","Gefitinib","AZD6244","AG1478","Lapatinib")],icbp_drug_er_neg[, 1:7],method="spearman",use="pairwise")

#ER Pos
pdf("~/Dropbox/Multipathway_profiling_paper/Genome_Medicine_Submission/Revison/Figures_Revised/DrugHeatMaps_ERSeperated.pdf", width=6)
heatmap.2(cors_drugs_er_pos, col=my_palette, trace='none', density.info = 'none', margins = c(6,6), cexCol = 0.8, main="ICBP (ER+ Only)",, RowSideColors = c("pink","yellow","yellow","yellow","pink","pink","pink","pink","lightblue", "pink","pink", "pink", "yellow","pink","pink","lightblue","lightblue","lightblue","lightblue", "pink"), scale="row",ColSideColors = c("coral3","aquamarine4","aquamarine4","coral3","coral3","aquamarine4","aquamarine4"))
#legend("bottomleft",legend = c("PI3K/AKT/mTOR inhibitors", "Chemotherapeutics","EGFR/MEK inhibitors","Growth phenotype","Survival phenotype"), col = c("pink","yellow","lightblue","coral3","aquamarine4"),  lty= 1,lwd = 10,cex = 1)
#ER neg
heatmap.2(cors_drugs_er_neg, col=my_palette, trace='none', density.info = 'none', margins = c(6,6), cexCol = 0.8, main="ICBP(ER- Only)", RowSideColors = c("pink","yellow","yellow","yellow","pink","pink","pink","pink","lightblue", "pink","pink", "pink", "yellow","pink","pink","lightblue","lightblue","lightblue","lightblue", "pink"), scale="row",ColSideColors = c("coral3","aquamarine4","aquamarine4","coral3","coral3","aquamarine4","aquamarine4"))
#legend("bottomleft",legend = c("PI3K/AKT/mTOR inhibitors", "Chemotherapeutics","EGFR/MEK inhibitors","Growth phenotype","Survival phenotype"), col = c("pink","yellow","lightblue","coral3","aquamarine4"),  lty= 1,lwd = 10,cex = 1)




```

##5(B): Correlate GFRN pathway predictions with drug response in an independent drug assay and make Heatmap
```{r, echo=FALSE,include=FALSE}
library(gplots)
# merge drug response data with GFRN pathway predictions
assay_ec50_preds<-merge_drop(assay_ec50,single_pathway_best_icbp)

cors_spearman=cor(assay_ec50_preds[,c(1,2,3,4,5,7,8,9)], assay_ec50_preds[,c(10:16)],  method="spearman", use="pairwise")
heatmap.2(cors_spearman,margins =c(10,10), col=my_palette, trace="none",main="",RowSideColors = c("pink","yellow","lightblue","pink","yellow","yellow","pink","lightblue"),ColSideColors = c("coral3","aquamarine4","aquamarine4","coral3","coral3","aquamarine4","aquamarine4"),density.info = 'none', scale="column")

assay_ec50_preds_subtype<-merge_drop(assay_ec50_preds, icbp_status)
assay_ec50_preds_subtype=assay_ec50_preds_subtype[,-21]
colnames(assay_ec50_preds_subtype)[13]="HER2"
indep_drug_er_pos=assay_ec50_preds_subtype[assay_ec50_preds_subtype$ER=="Positive",]
indep_drug_er_neg=assay_ec50_preds_subtype[assay_ec50_preds_subtype$ER=="Negative",]

cors_spearman_pos=cor(indep_drug_er_pos[,c(1,2,3,4,5,7,8,9)], indep_drug_er_pos[,c(10:16)],  method="spearman", use="pairwise")
cors_spearman_neg=cor(indep_drug_er_neg[,c(1,2,3,4,5,7,8,9)], indep_drug_er_neg[,c(10:16)],  method="spearman", use="pairwise")


heatmap.2(cors_spearman_pos, col=my_palette, trace="none",main="Independent Assay(ER+)",RowSideColors = c("pink","yellow","lightblue","pink","yellow","yellow","pink","lightblue"),ColSideColors = c("coral3","aquamarine4","aquamarine4","coral3","coral3","aquamarine4","aquamarine4"),density.info = 'none', scale="column")

heatmap.2(cors_spearman_neg, col=my_palette, trace="none",main="Independent Assay(ER-)",RowSideColors = c("pink","yellow","lightblue","pink","yellow","yellow","pink","lightblue"),ColSideColors = c("coral3","aquamarine4","aquamarine4","coral3","coral3","aquamarine4","aquamarine4"),density.info = 'none', scale="column")

dev.off()

```

###Supplemental Figures 3(A)-(H): 
Protein based pathway validations codes are in ./ASSIGN folder. Gene expression, mutation and IHC-based validations are shown here.
```{r,echo=FALSE,cache=TRUE}
#Gene expression based validation to check if predictions follow the main gene expression, ie, high prediction of a  pathway should have high expression of that pathway gene.

create_high_low_expression <-
  function(gene = NA,cutoff_percentile_high = 0.9,cutoff_percentile_low = 0.1)
  {
  exp <- test[gene,order(test[gene,])]
  cutoff_high <-
  quantile(t(test[gene,order(test[gene,])]),cutoff_percentile_high)
  cutoff_low <-
  quantile(t(test[gene,order(test[gene,])]),cutoff_percentile_low)
  
  gene_class <- NULL
  for (i in 1:length(exp)) {
  if (exp[i] >= cutoff_high)
  {
  gene_class[i] = "HIGH"
  }
  else if (exp[i] <= cutoff_low)
  {
  gene_class[i] = "LOW"
  }
  else if (cutoff_low<exp[i] & exp[i]<cutoff_high) 
  {
  gene_class[i] = "Intermediate"
  }
  }
  exp_class <- as.data.frame(cbind(t(exp),(gene_class)))
  rownames(exp_class) <- names(exp)
  colnames(exp_class)[2] <- paste(gene,"class",sep = "_")
  exp_class[,1] <- as.numeric(exp)
  return(exp_class)
  }
akt_class<-create_high_low_expression(gene = "AKT1",cutoff_percentile_high = .9,cutoff_percentile_low = 0.1)
boxplot2(merge_drop(akt_class,single_pathway_best_tcga)[,3]~merge_drop(akt_class,single_pathway_best_tcga)[,2],ylab="Estimated AKT activity")

egfr_class<-create_high_low_expression(gene = "EGFR",cutoff_percentile_high = .9,cutoff_percentile_low = 0.1)
boxplot2(merge_drop(egfr_class,single_pathway_best_tcga)[,5]~merge_drop(egfr_class,single_pathway_best_tcga)[,2],ylab="Estimated EGFR activity")

her2_class<-create_high_low_expression(gene = "ERBB2",cutoff_percentile_high = .66,cutoff_percentile_low = 0.1)
boxplot2(merge_drop(her2_class,single_pathway_best_tcga)[,6]~merge_drop(her2_class,single_pathway_best_tcga)[,2],ylab="Estimated HER2 activity")

igf1r_class<-create_high_low_expression(gene = "IGF1R",cutoff_percentile_high = .8,cutoff_percentile_low = 0.2)
boxplot2(merge_drop(igf1r_class,single_pathway_best_tcga)[,7]~merge_drop(igf1r_class,single_pathway_best_tcga)[,2],ylab="Estimated IGF1R activity")

raf1_class<-create_high_low_expression(gene = "RAF1",cutoff_percentile_high = .9,cutoff_percentile_low = 0.1)
boxplot2(merge_drop(raf1_class,single_pathway_best_tcga)[,9]~merge_drop(raf1_class,single_pathway_best_tcga)[,2],ylab="Estimated RAF1 activity")


##########
brca_mut<-t(read.table("Datasets/TCGA_PANCAN_mut.txt",header=1,row.names=1, sep='\t', check.names = F))
rownames(brca_mut)<- short_to_long_TCGA_id(longnames = colnames(test),shortnames = rownames(brca_mut))
pred_mut<- merge_drop(single_pathway_best_tcga,brca_mut)

boxplot2(pred_mut$AKT~pred_mut$PIK3CA,ylab="AKT estimated pathway",names=c("PIK3CA WT","PIK3CA mutated"),main=paste("p-value:",round(t.test(pred_mut$AKT~pred_mut$PIK3CA)$p.value,digits = 3),sep=" "))
boxplot2(pred_mut$BAD~pred_mut$PIK3CA,ylab="BAD estimated pathway",names=c("PIK3CA WT","PIK3CA mutated"),main=paste("p-value: ",t.test(pred_mut$BAD~pred_mut$PIK3CA)$p.value,sep=" "))

subtypes_preds_ihc<-merge(single_pathway_best_tcga,er_pr_her2_status,by.x = 0,by.y = 0,all.x =T)

# seperate into ER, PR, HER + and -
er_pos<-subtypes_preds_ihc[subtypes_preds_ihc$er_status_by_ihc=="Positive",]
er_neg<-subtypes_preds_ihc[subtypes_preds_ihc$er_status_by_ihc=="Negative",]
pr_pos<-subtypes_preds_ihc[subtypes_preds_ihc$pr_status_by_ihc=="Positive",]
pr_neg<-subtypes_preds_ihc[subtypes_preds_ihc$pr_status_by_ihc=="Negative",]
her_pos<-subtypes_preds_ihc[subtypes_preds_ihc$her2_status_by_ihc=="Positive",]
her_neg<-subtypes_preds_ihc[subtypes_preds_ihc$her2_status_by_ihc=="Negative",]
survival_phen<-single_pathway_best_tcga[single_pathway_best_tcga$phenotype=="Survival Phenotype",]
growth_phen<-single_pathway_best_tcga[single_pathway_best_tcga$phenotype=="Growth Phenotype",]

# boxplot her2+ vs her2 -
boxplot2(her_pos$HER2,her_neg$HER2,ylab="HER2 Activity",ylim=c(-3,3),las=1,names=c("HER2 +","HER2 -"),ylab="Estimated HER2 activity",main=paste("p-value: ",t.test(her_pos$HER2,her_neg$HER2)$p.value,sep=" "))
```

###Supplemental Figures 4(A)-(G): Boxplot GFRN pathway activity across breast cancer clinical subtypes (IHC)
```{r,echo=FALSE,cache=TRUE}
#merge subtypes with predictions
boxplot(er_pos$AKT,er_neg$AKT,pr_pos$AKT,pr_neg$AKT,her_pos$AKT,her_neg$AKT,survival_phen$AKT,growth_phen$AKT,las=2,at=c(1,2,4,5,7,8,10,11), names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main="ER/PR/HER2 status in association AKT Pathway", col=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"), cex.axis=1.2,ylab="Estimated AKT pathway activity")
boxplot(er_pos$BAD,er_neg$BAD,pr_pos$BAD,pr_neg$BAD,her_pos$BAD,her_neg$BAD,survival_phen$BAD,growth_phen$BAD,las=2,at=c(1,2,4,5,7,8,10,11), names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"),  main="ER/PR/HER2 status in association BAD Pathway",col=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"), cex.axis=1.2,ylab="Estimated BAD pathway activity")
boxplot(er_pos$HER2,er_neg$HER2,pr_pos$HER2,pr_neg$HER2,her_pos$HER2,her_neg$HER2, survival_phen$HER2,growth_phen$HER2,las=2,at=c(1,2,4,5,7,8,10,11), names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main="ER/PR/HER2 status in association HER2 Pathway",col=c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"), cex.axis=1.2,ylab="Estimated HER2 pathway activity")
boxplot(er_pos$IGF1R,er_neg$IGF1R,pr_pos$IGF1R,pr_neg$IGF1R,her_pos$IGF1R,her_neg$IGF1R,survival_phen$IGF1R,growth_phen$IGF1R, las=2,at=c(1,2,4,5,7,8,10,11), names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main="ER/PR/HER2 status in association IGF1R Pathway",col= c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"), cex.axis=1.2,ylab="Estimated IGF1R pathway activity")
boxplot(er_pos$EGFR,er_neg$EGFR,pr_pos$EGFR,pr_neg$EGFR,her_pos$EGFR,her_neg$EGFR,survival_phen$EGFR,growth_phen$EGFR, las=2,at=c(1,2,4,5,7,8,10,11), names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main="ER/PR/HER2 status in association EGFR Pathway",col= c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"), cex.axis=1.2, ylab="Estimated EGFR pathway activity")
boxplot(er_pos$KRASGV,er_neg$KRASGV,pr_pos$KRASGV,pr_neg$KRASGV,her_pos$KRASGV,her_neg$KRASGV,survival_phen$KRASGV,growth_phen$KRASGV,las=2,at=c(1,2,4,5,7,8,10,11), names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main="ER/PR/HER2 status in association KRAS Pathway",col= c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"), cex.axis=1.2, ylab="Estimated KRAS pathway activity")
boxplot(er_pos$RAF1,er_neg$RAF1,pr_pos$RAF1,pr_neg$RAF1,her_pos$RAF1,her_neg$RAF1,survival_phen$RAF1,growth_phen$RAF1, las=2,at=c(1,2,4,5,7,8,10,11), names=c("ER+","ER-","PR+","PR-","HER2+","HER2-","Survival","Growth"), main="ER/PR/HER2 status in association RAF1 Pathway",col= c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FFFF33","#F781BF","coral3","aquamarine4"), cex.axis=1.2,ylab="Estimated RAF1 pathway activity")



```

###Supplemental Figures 5(A)-(G): Boxplot GFRN pathway activity across breast cancer intrinsic subtypes (PAM50)
```{r, echo=FALSE,cache=TRUE}

# change TCGA BRCA names from short to long
rownames(pam50)<-short_to_long_TCGA_id(longnames = colnames(test),shortnames = gsub(pattern = "\\.",replacement = "-",rownames(pam50)))
pam50_subset=subset(pam50, pam50$PAM50!="Normal")
pam50_com_tcga<-merge_drop(pca_mat$x,pam50_subset)

subtypes_preds<-merge_drop(single_pathway_best_tcga,pam50_subset)
basal<-subtypes_preds[subtypes_preds$PAM50=="Basal",]
her2<-subtypes_preds[subtypes_preds$PAM50=="Her2",]
lumA<-subtypes_preds[subtypes_preds$PAM50=="LumA",]
lumB<-subtypes_preds[subtypes_preds$PAM50=="LumB",]

# boxplots for GFRN activity across intrinistic subtypes
par(mfrow=c(1,1),lwd=3)
boxplot(basal$AKT,her2$AKT,lumA$AKT,lumB$AKT,col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'),ylab="AKT Activity",names = c("Basal","Her2","LumA","LumB"),main="Intrinsic subtypes in association AKT Pathway")
boxplot(basal$BAD,her2$BAD,lumA$BAD,lumB$BAD,col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'),ylab="BAD Activity",names = c("Basal","Her2","LumA","LumB"),las=1,main="Intrinsic subtypes in association BAD Pathway")
boxplot(basal$EGFR,her2$EGFR,lumA$EGFR,lumB$EGFR,col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'),ylab="EGFR Activity",las=1,names = c("Basal","Her2","LumA","LumB"),main="Intrinsic subtypes in association EGFR Pathway")
boxplot(basal$HER2,her2$HER2,lumA$HER2,lumB$HER2,col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'),ylab="HER2 Activity",las=1,names = c("Basal","Her2","LumA","LumB"),main="Intrinsic subtypes in association HER2 Pathway")
boxplot(basal$IGF1R,her2$IGF1R,lumA$IGF1R,lumB$IGF1R,col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'),ylab="IGF1R Activity",las=1,names = c("Basal","Her2","LumA","LumB"),main="Intrinsic subtypes in association IGF1R Pathway")
boxplot(basal$KRAS,her2$KRAS,lumA$KRAS,lumB$KRAS,col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'),ylab="KRAS Activity",las=1,names = c("Basal","Her2","LumA","LumB"),main="Intrinsic subtypes in association KRAS Pathway")
boxplot(basal$RAF1,her2$RAF1,lumA$RAF1,lumB$RAF1,col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'),ylab="RAF1 Activity",las=1,names = c("Basal","Her2","LumA","LumB"),main="Intrinsic subtypes in association RAF1 Pathway")


```

###Supplemental Figure 6(A)&(B): Boxplot TCGA Principal Components across BRCA intrinsic subtypes (PAM50)
```{r, echo=FALSE}
# Take the mean of all the gene expression values for all samples and correlate with the PCs 
par(mfrow=c(1,1),lwd=4)
##Supplemental Figure 6(A) 
plot(pca_mat$x[,1],(apply(test_f,2,mean)), main="Average expression vs PC1\n in TCGA samples", ylab = "Mean Gene Expression (RNA-Seq(TPMlog2)", xlab="PCA Values")
print(paste("Spearman correlation of PC1 and mean gene expression of each sample: ",cor(pca_mat$x[,1],(apply(test_f,2,mean)), method= "spearman"),sep=" ")) # -0.823

# PCA analysis of intrinstic subtypes in TCGA BRCA

# boxplot for PCs in TCGA BRCA across subtypes
boxplot(pam50_com_tcga$PC1~as.matrix(pam50_com_tcga$PAM50),main="PC1 across intrinsic subtypes",cex=0.8, col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'), cex.axis=1.1,ylab="PC1 values")
boxplot(pam50_com_tcga$PC2~as.matrix(pam50_com_tcga$PAM50),main="PC2 across intrinsic subtypes",cex=0.8, col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'), cex.axis=1.1,ylab="PC2 values")
boxplot(pam50_com_tcga$PC3~as.matrix(pam50_com_tcga$PAM50),main="PC3 across intrinsic subtypes",cex=0.8, col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'),cex.axis=1.1,ylab="PC3 values")
boxplot(pam50_com_tcga$PC4~as.matrix(pam50_com_tcga$PAM50),main="PC4 across intrinsic subtypes",cex=0.8, col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'),cex.axis=1.1,ylab="PC4 values")
boxplot(pam50_com_tcga$PC5~as.matrix(pam50_com_tcga$PAM50),main="PC5 across intrinsic subtypes",cex=0.8, col=c('#e41a1c',brewer.pal(6, "Dark2")[4],brewer.pal(6, "Dark2")[5],'#fccaa4'),cex.axis=1.1,ylab="PC5 values")

pca_kmeans<-merge_drop(pam50_com_tcga,single_pathway_best_tcga)
boxplot(pca_kmeans$PC1~as.matrix(pca_kmeans$kmeans.cluster),las=2,main="PC1 across subgroups",cex=0.5, col=c(brewer.pal(5, "Set1")[3],brewer.pal(5, "Set1")[5],brewer.pal(5, "Set1")[1],brewer.pal(5, "Set1")[2]), cex.axis=0.5,ylab="PC1 values")

boxplot(pca_kmeans$PC2~as.matrix(pca_kmeans$kmeans.cluster),las=2,main="PC2 across subgroups",cex=0.5, col=c(brewer.pal(5, "Set1")[3],brewer.pal(5, "Set1")[5],brewer.pal(5, "Set1")[1],brewer.pal(5, "Set1")[2]), cex.axis=0.5,ylab="PC2 values")

boxplot(pca_kmeans$PC3~as.matrix(pca_kmeans$kmeans.cluster),las=2,main="PC3 across subgroups",cex=0.5, col=c(brewer.pal(5, "Set1")[3],brewer.pal(5, "Set1")[5],brewer.pal(5, "Set1")[1],brewer.pal(5, "Set1")[2]), cex.axis=0.5,ylab="PC3 values")

boxplot(pca_kmeans$PC4~as.matrix(pca_kmeans$kmeans.cluster),las=2,main="PC4 across subgroups",cex=0.5, col=c(brewer.pal(5, "Set1")[3],brewer.pal(5, "Set1")[5],brewer.pal(5, "Set1")[1],brewer.pal(5, "Set1")[2]), cex.axis=0.5,ylab="PC4 values")

boxplot(pca_kmeans$PC5~as.matrix(pca_kmeans$kmeans.cluster),las=2,main="PC5 across subgroups",cex=0.5, col=c(brewer.pal(5, "Set1")[3],brewer.pal(5, "Set1")[5],brewer.pal(5, "Set1")[1],brewer.pal(5, "Set1")[2]), cex.axis=0.5,ylab="PC5 values")

```

###Supplemental Figures 7(A)-(D):
```{r,cache=TRUE,echo=FALSE,warning=FALSE}

rownames(drugs)[38] <- "SUM1315"
rownames(drugs)[45] <- "T47D.Kbluc"
rownames(drugs)[52] <- "MDAMB175"

drugs_all <- drugs[rownames(single_pathway_best_icbp),11:100]

drugs <- drugs[rownames(single_pathway_best_icbp),c("GSK2141795","PF.4691502",
                                                    "Everolimus","Sigma.AKT1.2.inhibitor",
                                                    "GSK1059868","Rapamycin",
                                                    "Temsirolimus","GSK2126458",
                                                    "GSK2119563","GSK1059615",
                                                    "Lapatinib","AG1478",
                                                    "Gefitinib","AZD6244",
                                                    "Erlotinib","GSK1120212",
                                                    "Docetaxel","CGC.11047",
                                                    "Paclitaxel","Cisplatin")]
#Perform k-means clustering with 4 clusters
set.seed(123)
kmeans.4 <- kmeans(scale(single_pathway_best_icbp)[,c(1,2,3,4)], 4,
                   iter.max=100000, nstart = 100)
kmeans.4.clusters <-data.frame(kmeans.4$cluster)
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "3",] <- "EGFR/BAD low"
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "1",] <- "EGFR/BAD high"
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "4",] <- "AKT/HER2 high"
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "2",] <- "AKT/HER2 low"
colnames(kmeans.4.clusters) <- "KMeans.Clusters"

ttest_boxplots <- function(drugdata, clusters, selected_clustera, selected_clusterb, selected_drug){
  numa <- sum(!is.na(drugdata[rownames(clusters)[clusters == selected_clustera], selected_drug]))
  numb <- sum(!is.na(drugdata[rownames(clusters)[clusters == selected_clusterb], selected_drug]))
  if(numa < 2 | numb < 2){
    return(NA)
  }
  result <- t.test(drugdata[rownames(clusters)[clusters == selected_clustera], selected_drug],
                   drugdata[rownames(clusters)[clusters == selected_clusterb], selected_drug], na.rm=T)
  return(result$p.value)
}

akt.ttest.results <- unlist(lapply(colnames(drugs), function(x) ttest_boxplots(drugs, kmeans.4.clusters, "AKT/HER2 high", "AKT/HER2 low", x)))
akt.ttest.results.fdr <- p.adjust(akt.ttest.results, method="fdr")

egfr.ttest.results <- unlist(lapply(colnames(drugs), function(x) ttest_boxplots(drugs, kmeans.4.clusters, "EGFR/BAD high", "EGFR/BAD low", x)))
egfr.ttest.results.fdr <- p.adjust(egfr.ttest.results, method="fdr")

ttest_results <- data.frame(akt.ttest.results=akt.ttest.results,
                            akt.ttest.results.fdr=akt.ttest.results.fdr,
                            egfr.ttest.results=egfr.ttest.results,
                            egfr.ttest.results.fdr=egfr.ttest.results.fdr,
                            row.names=colnames(drugs))

#Status API Training Shop Blog About 
set.seed(123)
kmeans.4 <- kmeans(scale(single_pathway_best_icbp)[,c(1,2,3,4)], 4, iter.max=100000, nstart = 100)
kmeans.4.clusters <-data.frame(kmeans.4$cluster)
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "3",] <- "Growth\nBAD low"
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "1",] <- "Growth\nBAD high"
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "4",] <- "Survival\nHER2 high"
kmeans.4.clusters[kmeans.4.clusters$kmeans.4.cluster == "2",] <- "Survival\nHER2 low"
colnames(kmeans.4.clusters) <- "KMeans.Clusters"
kmeans.4.clusters <- data.frame(kmeans.4.clusters$KMeans.Clusters, row.names = rownames(kmeans.4.clusters) )

pathway_colors <- brewer.pal(5, "Set1")[c(1,2,3,5)]

plot_drug_boxplots_final <- function(drugdata, clusters, selected, fillcolor){
  colnames(clusters) <- "clusters"
  merged <- merge(drugdata, clusters, by=0, all=T)
  merged_melt <- melt(merged, id.vars = c("Row.names", "clusters"))
  merged_melt_s <- merged_melt[merged_melt$variable == selected,]
  theplot <- ggplot(merged_melt_s, aes(factor(clusters, 
                                              levels = c("Survival\nHER2 high",
                                                         "Survival\nHER2 low",
                                                         "Growth\nBAD high",
                                                         "Growth\nBAD low")), value)) +
    geom_boxplot(outlier.shape = NA, fill=fillcolor) + 
    theme_classic() +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
          axis.line.x = element_line(colour = 'black', size=0.5, linetype='solid'),
          axis.line.y = element_line(colour = 'black', size=0.5, linetype='solid')) +
    xlab("") +
    ylab("-log(EC50)") +
    geom_point(position=position_jitter(width = 0.1)) +
    ggtitle(selected) 
  return(theplot)
}

plota <- plot_drug_boxplots_final(drugs, kmeans.4.clusters, "GSK1059615", pathway_colors)
plotb <- plot_drug_boxplots_final(drugs, kmeans.4.clusters, "Lapatinib", pathway_colors)
plotc <- plot_drug_boxplots_final(drugs, kmeans.4.clusters, "AG1478", pathway_colors)
plotd <- plot_drug_boxplots_final(drugs, kmeans.4.clusters, "Gefitinib", pathway_colors)


grid.arrange(plota,plotb,plotc,plotd, nrow=2,ncol=2)



```
###Supplemental Figure 9:Survival analysis with TCGA BRCA data...
```{r}
library(plyr)
surv<-read.table("Datasets/TCGA_BRCA_survival_tcga.txt",header=1,row.names=1,sep='\t',stringsAsFactors = T,skip = 1)
rownames(surv)<-short_to_long_TCGA_id(longnames =rownames(pam50),shortnames = rownames(surv) )
pam50_surv<-merge_drop(pam50,surv)
pam50_surv_tcga_4_clusters<-merge_drop(pam50_surv,single_pathway_best_tcga)
#pam50_surv_tcga_4_clusters$phenotype<-gsub(pattern = "/.*",replacement = "",pam50_surv_tcga_4_clusters$KMeans.Clusters)

fit<-survfit(Surv(pam50_surv_tcga_4_clusters$OS.Time,pam50_surv_tcga_4_clusters$OS.event)~pam50_surv_tcga_4_clusters$phenotype)
plot(fit,col = c("aquamarine4","coral3"), xlab="Time in days",ylab="Survival probability",main="Survival in TCGA BRCA\n patients based on 2 phenotypes",lwd = 4)
legend("bottomleft",c("Growth phenotype","Survival phenotype"),col=c("aquamarine4","coral3"),lwd=4,bty="n",cex=0.9)
diff<-survdiff(Surv(pam50_surv_tcga_4_clusters$OS.Time,pam50_surv_tcga_4_clusters$OS.event)~pam50_surv_tcga_4_clusters$phenotype,rho=1)# rho=1-->Peto & Peto modification of the Gehan-Wilcoxon test; rho = 0 -->log-rank or Mantel-Haenszel test
diff

text(800,0.2,paste("   p-value= ",round(1-pchisq(diff$chisq, length(diff$n)- 1),digit=2)),adj = c(0,0))

fit<-survfit(Surv(pam50_surv_tcga_4_clusters$OS.Time,pam50_surv_tcga_4_clusters$OS.event)~pam50_surv_tcga_4_clusters$kmeans.cluster)
plot(fit,col=c(brewer.pal(5, "Set1")[3],brewer.pal(5, "Set1")[5],brewer.pal(5, "Set1")[1],brewer.pal(5, "Set1")[2]), xlab="Time in days",ylab="Survival probability",main="Survival in TCGA BRCA\n patients based on 4 subgroups",lwd = 4)
legend("bottomleft",c("Growth/BAD High","Growth/BAD Low","Survival/HER2 High","Survival/HER2 Low"),bty="n" ,col=c(brewer.pal(5, "Set1")[3],brewer.pal(5, "Set1")[5],brewer.pal(5, "Set1")[1],brewer.pal(5, "Set1")[2]),lwd=4,cex=0.9)
diff<-survdiff(Surv(pam50_surv_tcga_4_clusters$OS.Time,pam50_surv_tcga_4_clusters$OS.event)~pam50_surv_tcga_4_clusters$kmeans.cluster, rho = 1)
diff

text(800,0.2,paste("   p-value= ",round(1-pchisq(diff$chisq, length(diff$n)- 1),digit=2)),adj = c(0,0))

# 
# fit<-survfit(Surv(pam50_surv_tcga_4_clusters$OS.Time,pam50_surv_tcga_4_clusters$OS.event)~pam50_surv_tcga_4_clusters$ER.Status)
# plot(fit,col = 2:6, xlab="Time in days",ylab="Survival probability",main="Survival in TCGA BRCA\n patients based on ER status",lwd = 4)
# legend("topright",c("Indeterminate","Negative","Not Performed","Performed but unavailable","Positive"),col=2:6,lwd=4,box.lwd = 1)
# survdiff(Surv(pam50_surv_tcga_4_clusters$OS.Time,pam50_surv_tcga_4_clusters$OS.event)~pam50_surv_tcga_4_clusters$ER.Status,rho=0)
# 
# fit<-survfit(Surv(pam50_surv_tcga_4_clusters$OS.Time,pam50_surv_tcga_4_clusters$OS.event)~pam50_surv_tcga_4_clusters$PR.Status)
# plot(fit,col = 2:6, xlab="Time in days",ylab="Survival probability",main="Survival in TCGA BRCA\n patients based on PR status",lwd = 4)
# legend("topright",c("Indeterminate","Negative","Not Performed","Performed but unavailable","Positive"),col=2:6,lwd=4,box.lwd = 1)
# survdiff(Surv(pam50_surv_tcga_4_clusters$OS.Time,pam50_surv_tcga_4_clusters$OS.event)~pam50_surv_tcga_4_clusters$PR.Status,rho=0)
# 
# 
# #pam50_surv_tcga_4_clusters_filtered<-pam50_surv_tcga_4_clusters[pam50_surv_tcga_4_clusters$PAM50!="Normal",]
# fit<-survfit(Surv(pam50_surv_tcga_4_clusters$OS.Time,pam50_surv_tcga_4_clusters$OS.event)~pam50_surv_tcga_4_clusters$PAM50)
# plot(fit,col = 2:6, xlab="Time in days",ylab="Survival probability",main="Survival in TCGA BRCA\n patients based on PAM50",lwd = 4)
# legend("topright",c("Basal","HER2","LumA","LumB","Normal"),col=2:6,lwd=4,box.lwd = 1)
# survdiff(Surv(pam50_surv_tcga_4_clusters$OS.Time,pam50_surv_tcga_4_clusters$OS.event)~pam50_surv_tcga_4_clusters$PAM50,rho=0)
# summary(fit)


tcga_clinicals<-merge_drop(single_pathway_best_tcga,t(clinicals[c("vital_status","last_contact_days_to","death_days_to"),3:ncol(clinicals)]))
#tcga_clinicals<-merge_drop(single_pathway_best_tcga_scaled,t(clinicals[c("vital_status","last_contact_days_to","death_days_to"),3:ncol(clinicals)]))
tcga_clinicals$vital_status<-factor(tcga_clinicals$vital_status)
tcga_clinicals$vital_status<-mapvalues(tcga_clinicals$vital_status,from=c("Alive","Dead"),to=c(0,1))
tcga_clinicals$last_contact_days_to<-as.numeric(as.character(tcga_clinicals$last_contact_days_to))
tcga_clinicals$vital_status<-as.numeric(as.character(tcga_clinicals$vital_status))
tcga_clinicals$death_days_to<-as.numeric(as.character(tcga_clinicals$death_days_to))
tcga_clinicals<-tcga_clinicals[!is.na(tcga_clinicals$vital_status),]
tcga_clinicals$time=NULL
for(i in 1: nrow(tcga_clinicals)){
  if(tcga_clinicals$vital_status[i]==1){
    tcga_clinicals$time[i]<-tcga_clinicals$death_days_to[i]
  }
  else{
    tcga_clinicals$time[i]<-tcga_clinicals$last_contact_days_to[i]  
  }
  
    
}
tcga_clinicals<-tcga_clinicals[!tcga_clinicals$time<0,]
#tcga_clinicals$phenotype<-gsub(pattern = "/.*",replacement = "",tcga_clinicals$KMeans.Clusters)
fit<-survfit(Surv(tcga_clinicals$time,tcga_clinicals$vital_status)~as.factor(tcga_clinicals$phenotype))

plot(fit,col=c("aquamarine4","coral3") , xlab="Time in days",ylab="Survival probability",main="",lwd = 4)
legend("bottomleft",c("Growth phenotype","Survival phenotype"),col=c("aquamarine4","coral3"),lwd=4,bty="n",cex=0.9)
diff<-survdiff(Surv(tcga_clinicals$time,tcga_clinicals$vital_status)~as.factor(tcga_clinicals$phenotype),rho=1)# rho=1-->Peto & Peto modification of the Gehan-Wilcoxon test; rho = 0 -->log-rank or Mantel-Haenszel test
diff

text(800,0.2,paste("   p-value= ",round(1-pchisq(diff$chisq, length(diff$n)- 1),digit=2)),adj = c(0,0))

fit<-survfit(Surv(tcga_clinicals$time,tcga_clinicals$vital_status)~tcga_clinicals$kmeans.cluster)
plot(fit,col = c(brewer.pal(5, "Set1")[3],brewer.pal(5, "Set1")[5],brewer.pal(5, "Set1")[1],brewer.pal(5, "Set1")[2]), xlab="Time in days",ylab="Survival probability",main="Survival in TCGA BRCA\n patients based on 4 subgroups",lwd = 4)
legend("bottomleft",c("Growth/BAD high", "Growth/BAD low", "Survival/HER2 high", "Survival/HER2 low"),col=c(brewer.pal(5, "Set1")[3],brewer.pal(5, "Set1")[5],brewer.pal(5, "Set1")[1],brewer.pal(5, "Set1")[2]),lwd=4,bty = "n",cex=0.9)
diff<-survdiff(Surv(tcga_clinicals$time,tcga_clinicals$vital_status)~tcga_clinicals$kmeans.cluster,rho=1)
diff

text(800,0.2,paste("   p-value= ",round(1-pchisq(diff$chisq, length(diff$n)- 1),digit=2)),adj = c(0,0))


```


```{r, echo=FALSE}
time<-format(Sys.time(),"%a %b %d %X %Y")
```
This analysis was run on `r time` 